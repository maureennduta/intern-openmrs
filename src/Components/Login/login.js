import {
    Form,
    TextInput,
    Button,
    InlineNotification,
  } from "carbon-components-react";
  import React, { useState } from "react";
  import { LoginUser } from "../Auth/Auth";
  import { useHistory } from "react-router-dom";
  import "./login.scss";
  const btoa = require("btoa");
  
  const Login = ({ setIsAuthenticated }) => {
    const [Username, setUsername] = useState("");
    const [Password, setPassword] = useState("");
    const [error, setError] = useState("");
    const history = useHistory();
  
    const HandleAuth = async (e) => {
      e.preventDefault();
  
      const token = btoa(`${Username}:${Password}`);
      await LoginUser(token).then((resp) => {
        if (resp.authenticated === true) {
          setIsAuthenticated(true);
          sessionStorage.setItem("user", token);
          history.push("/PatientsRecords");
        } else {
          setError("Wrong username or password");
        }
      });
    };
  
    return (
        <Form onSubmit={HandleAuth} data-testid="login">
          <div className="bx--grid--full-width">
            <div className="bx--row">
              <div className="bx--col col-sm-2"></div>
              <div className="bx--col bx--col-md-6 col-sm" id="loginform">
                <h1>Welcome to OPENMRS</h1>
                <hr />
                <br />
  
                <TextInput
                  labelText="User name: "
                  id="name"
                  data-testid="name"
                  required
                  placeholder="Enter user name"
                  onChange={(e) => setUsername(e.target.value)}
                />
                <TextInput.PasswordInput
                  id="password"
                  data-testid="password"
                  hidePasswordLabel="Hide password"
                  invalidText="A valid value is required"
                  labelText="Password: "
                  required
                  placeholder="Enter Password"
                  showPasswordLabel="Show password"
                  onChange={(e) => setPassword(e.target.value)}
                />
                <br />
                {error && (
                  <div>
                    <InlineNotification
                      className="notification--error"
                      kind="error"
                      iconDescription="describes the close button"
                      subtitle={
                        <span>
                          <h6>{error}</h6>
                        </span>
                      }
                      title="Error!!"
                    />
                  </div>
                )}
                <Button
                  size="default"
                  kind="primary"
                  type="submit"
                  data-testid="submit"
                >
                  Login
                </Button>
                
              </div>
              <div className="bx--col col-sm-2"></div>
            </div>
          </div>
        </Form>
      
    );
  };
  export default Login;