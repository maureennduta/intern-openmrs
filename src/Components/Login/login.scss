@import "carbon-components/scss/globals/grid/grid";
#loginform {
  margin-top: 10%;
  background-color: rgb(255, 253, 251);
  font-weight: bold;
  border: 3px solid #f1f1f1;
  height: 50vh;
  border-radius: 1px;
  	box-shadow: 0 14px 28px rgba(0,0,0,0.25), 
			0 10px 10px rgba(0,0,0,0.22);
	position: relative;
  padding: 1rem;
}
#name{
  margin-bottom: 2rem;
}
