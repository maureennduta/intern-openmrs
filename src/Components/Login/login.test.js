import React from 'react';
import ReactDOM from 'react-dom'; // use this to render
import Login from './login';
import {render,cleanup} from '@testing-library/react'; //this render diff from reactdom render
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
// import { rest } from 'msw';//these two mock an API request that our tested component makes.
// import { setupServer } from 'msw/node';
// import Axios from 'axios';

//cleanup cleans tests for you
afterEach(cleanup);
it("renders without crashing", ()=> {
    const div = document.createElement("div");
    ReactDOM.render(<Login></Login>,div)
})
//passed

it("renders form correctly",() =>{
    const {getByTestId} = render(<Login></Login>)
    expect(getByTestId("name")).toBeRequired()
})
//toBeRequired checks if an element is required
//passed

// test('login',() => {
//     loadLogin().then( =>{
//         expect(loginform).toBeEnabled();
//     })
// });

it("matches snapshot",() => {
    const tree = renderer.create(<Login></Login>).toJSON();
    expect(tree).toMatchSnapshot();
})
//snapshot testing useful for making sure UI doesnt change unexpectedly
//snapshot testing is provided by Jest and verifies the component rendering result
