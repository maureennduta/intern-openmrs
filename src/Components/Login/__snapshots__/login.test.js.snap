// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  className="bx--form"
  data-testid="login"
  onSubmit={[Function]}
>
   
  <div
    className="bx--grid--full-width"
  >
    <div
      className="bx--row"
    >
      <div
        className="bx--col col-sm-2"
      />
      <div
        className="bx--col bx--col-md-6 col-sm"
        id="loginform"
      >
        <h1>
          Welcome to OPENMRS
        </h1>
        <hr />
        <br />
        <div
          className="bx--form-item bx--text-input-wrapper"
        >
          <label
            className="bx--label"
            htmlFor="name"
          >
            User name: 
          </label>
          <div
            className="bx--text-input__field-outer-wrapper"
          >
            <div
              className="bx--text-input__field-wrapper"
              data-invalid={null}
            >
              <input
                className="bx--text-input bx--text__input"
                data-testid="name"
                disabled={false}
                id="name"
                onChange={[Function]}
                onClick={[Function]}
                placeholder="Enter user name"
                required={true}
                title="Enter user name"
                type="text"
              />
            </div>
          </div>
        </div>
        <div
          className="bx--form-item bx--text-input-wrapper bx--password-input-wrapper"
        >
          <label
            className="bx--label"
            htmlFor="password"
          >
            Password: 
          </label>
          <div
            className="bx--text-input__field-outer-wrapper"
          >
            <div
              className="bx--text-input__field-wrapper"
              data-invalid={null}
            >
              <input
                className="bx--text-input bx--password-input \${prefix}--text__input"
                data-testid="password"
                data-toggle-password-visibility={true}
                disabled={false}
                id="password"
                onChange={[Function]}
                onClick={[Function]}
                placeholder="Enter Password"
                required={true}
                type="password"
              />
              <button
                className="bx--text-input--password__visibility__toggle bx--btn bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--bottom bx--tooltip--align-center"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <span
                  className="bx--assistive-text"
                >
                  Show password
                </span>
                <svg
                  aria-hidden={true}
                  className="bx--icon-visibility-on"
                  fill="currentColor"
                  focusable="false"
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15.5,7.8C14.3,4.7,11.3,2.6,8,2.5C4.7,2.6,1.7,4.7,0.5,7.8c0,0.1,0,0.2,0,0.3c1.2,3.1,4.1,5.2,7.5,5.3	c3.3-0.1,6.3-2.2,7.5-5.3C15.5,8.1,15.5,7.9,15.5,7.8z M8,12.5c-2.7,0-5.4-2-6.5-4.5c1-2.5,3.8-4.5,6.5-4.5s5.4,2,6.5,4.5	C13.4,10.5,10.6,12.5,8,12.5z"
                  />
                  <path
                    d="M8,5C6.3,5,5,6.3,5,8s1.3,3,3,3s3-1.3,3-3S9.7,5,8,5z M8,10c-1.1,0-2-0.9-2-2s0.9-2,2-2s2,0.9,2,2S9.1,10,8,10z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <br />
        
        <button
          aria-describedby={null}
          aria-pressed={null}
          className="bx--btn bx--btn--primary"
          data-testid="submit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          tabIndex={0}
          type="submit"
        >
          Login
        </button>
      </div>
      <div
        className="bx--col col-sm-2"
      />
    </div>
  </div>
   
</form>
`;
